Fails to configure in git worktree

```
$ git worktree add ../branch
$ cd ../branch
$ ./gradlew
```

```
FAILURE: Build failed with an exception.

* What went wrong:
Could not create task ':installKotlinterPrePushHook'.
> Could not create task of type 'InstallPrePushHookTask'.
   > Not a directory
```

as https://github.com/jeremymailen/kotlinter-gradle/blob/4f6944a0ca8eba4573467c24c52515b5303d5772/src/main/kotlin/org/jmailen/gradle/kotlinter/tasks/GitHookTasks.kt#L40 tries to `mkdirs()` immediately upon configuration, but in a [git worktree](https://git-scm.com/docs/git-worktree), `.git` is actually a text file that points to the shared GIT_DIR.

--- a/README.md
+++ b/README.md
@@ -19,7 +19,7 @@ Available on the Gradle Plugins Portal: https://plugins.gradle.org/plugin/org.jm
 
 ```kotlin
 plugins {
-    id("org.jmailen.kotlinter") version "3.1.0"
+    id("org.jmailen.kotlinter") version "3.1.1"
 }
 ```
 
@@ -30,7 +30,7 @@ plugins {
 
 ```groovy
 plugins {
-    id "org.jmailen.kotlinter" version "3.1.0"
+    id "org.jmailen.kotlinter" version "3.1.1"
 }
 ```
 
@@ -50,7 +50,7 @@ buildscript {
         }
     }
     dependencies {
-        classpath("org.jmailen.gradle:kotlinter-gradle:3.1.0")
+        classpath("org.jmailen.gradle:kotlinter-gradle:3.1.1")
     }
 }
 ```
@@ -75,7 +75,7 @@ buildscript {
         }
     }
     dependencies {
-        classpath "org.jmailen.gradle:kotlinter-gradle:3.1.0"
+        classpath "org.jmailen.gradle:kotlinter-gradle:3.1.1"
     }
 }
 ```

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -5,7 +5,7 @@ plugins {
     id("com.gradle.plugin-publish") version "0.12.0"
     `java-gradle-plugin`
     `maven-publish`
-    id("org.jmailen.kotlinter") version "3.0.1"
+    id("org.jmailen.kotlinter") version "3.1.0"
     idea
 }
 
@@ -19,7 +19,7 @@ val githubUrl ="https://github.com/jeremymailen/kotlinter-gradle"
 val webUrl = "https://github.com/jeremymailen/kotlinter-gradle"
 val projectDescription = "Lint and formatting for Kotlin using ktlint with configuration-free setup on JVM and Android projects"
 
-version = "3.1.0"
+version = "3.1.1"
 group = "org.jmailen.gradle"
 description = projectDescription
 

--- a/src/main/kotlin/org/jmailen/gradle/kotlinter/tasks/GitHookTasks.kt
+++ b/src/main/kotlin/org/jmailen/gradle/kotlinter/tasks/GitHookTasks.kt
@@ -1,6 +1,7 @@
 package org.jmailen.gradle.kotlinter.tasks
 
 import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.Internal
 import org.gradle.api.tasks.TaskAction
 import org.jmailen.gradle.kotlinter.support.VersionProperties
@@ -36,25 +37,23 @@ open class InstallPrePushHookTask : InstallHookTask("pre-push") {
 /**
  * Install or update a kotlinter-gradle hook.
  */
-abstract class InstallHookTask(hookFileName: String) : DefaultTask() {
-    @Internal val hooksDir = project.rootProject.file(".git/hooks").apply { mkdirs() }
-    @Internal val hookFile = File(hooksDir, hookFileName).apply {
-        createNewFile().and(setExecutable(true))
-    }
-    @get:Internal
-    val hookFileContent by lazy { hookFile.readText() }
+abstract class InstallHookTask(@get:Internal val hookFileName: String) : DefaultTask() {
+    @Input
+    val gitDirPath = property(default = ".git")
 
     @get:Internal
     abstract val hookContent: String
 
     init {
         outputs.upToDateWhen {
-            hookFileContent.contains(hookVersion)
+            getHookFile()?.readText()?.contains(hookVersion) ?: false
         }
     }
 
     @TaskAction
     fun run() {
+        val hookFile = getHookFile(true) ?: return
+        val hookFileContent = hookFile.readText()
 
         if (hookFileContent.isEmpty()) {
             logger.info("creating hook file: $hookFile")
@@ -77,6 +76,23 @@ abstract class InstallHookTask(hookFileName: String) : DefaultTask() {
         }
 
         logger.quiet("Wrote hook to $hookFile")
+    }
+
+    private fun getHookFile(warn: Boolean = false): File? {
+        val gitDir = project.rootProject.file(gitDirPath.get())
+        if (!gitDir.isDirectory) {
+            if (warn) logger.warn("skipping hook creation because $gitDir is not a directory")
+            return null
+        }
+        return try {
+            val hooksDir = File(gitDir, "hooks").apply { mkdirs() }
+            File(hooksDir, hookFileName).apply {
+                createNewFile().and(setExecutable(true))
+            }
+        } catch (e: Exception) {
+            if (warn) logger.warn("skipping hook creation because could not create hook under $gitDir: ${e.message}")
+            null
+        }
     }
 
     private val gradleCommand: String by lazy {

--- a/src/test/kotlin/org/jmailen/gradle/kotlinter/functional/InstallHookTaskTest.kt
+++ b/src/test/kotlin/org/jmailen/gradle/kotlinter/functional/InstallHookTaskTest.kt
@@ -104,6 +104,7 @@ abstract class InstallHookTaskTest(
 
     @Test
     fun `up-to-date when after hook installed`() {
+        File(testProjectDir.root, ".git").apply { mkdir() }
         lateinit var hookContent: String
         build(taskName).apply {
             assertEquals(SUCCESS, task(":$taskName")?.outcome)

